File: python/greedy/128.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        count = 1
        result = 0
        map = {}
        # hashset 存储所有元素
        for num in nums:
            map[num] = True

        for num in map:
            if num-1 in map:
                continue
            cur = num+1
            while cur in map:
                cur += 1
                count += 1
            result = max(result,count)
            count = 1

        return result

File: Java/greedy/128.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int result = 0;
        int count = 1;
        HashSet<Integer> cur = new HashSet<>();
        for (int num: nums) {
            cur.add(num);
        }
        for (Integer value: cur) {
            if (cur.contains(value-1)){
                continue;
            }
            int cur_value = value+1;
            while (cur.contains(cur_value)){
                cur_value++;
                count++;
            }
            if (count>result){
                result=count;
            }
            count=1;
        }
        return nums.length==0? 0:Math.max(count,result);
    }
}